it/google-cloud-platform/src/test/java/com/google/cloud/teleport/it/gcp/bigtable/BigTableIOLT.java

```diff
@@ -0,0 +1,250 @@
+ * -DfailIfNoTests=false".
+ *
+ * <p>Example trigger command for specific test: "mvn test -pl it/google-cloud-platform -am \
+ * -Dtest="BigTableIOLT#testAvroFileLoadsWriteThenRead" -Dconfiguration=local -Dproject=[gcpProject]
+ * \ -DartifactBucket=[temp bucket] -DfailIfNoTests=false".
+ */
+public class BigTableIOLT extends IOLoadTestBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(BigTableIOLT.class);
+  private static final String COLUMN_FAMILY_NAME = "cf";
+
+  private static BigtableResourceManager resourceManager;
+  private static final String READ_ELEMENT_METRIC_NAME = "read_count";
+  private Configuration configuration;
+  private BigtableOptions bigtableOptions;
+  private String tableId;
+
+  @Rule public TestPipeline writePipeline = TestPipeline.create();
+  @Rule public TestPipeline readPipeline = TestPipeline.create();
+
+  @Before
+  public void setup() throws IOException {
+    resourceManager = DefaultBigtableResourceManager.builder(testName, project).build();
+
+    bigtableOptions =
+        BigtableOptions.builder()
+            .setProjectId(project)
+            .setInstanceId(resourceManager.getInstanceId())
+            .setUserAgent("apache-beam-test")
+            .build();
+
+    String testConfig =
+        TestProperties.getProperty("configuration", "medium", TestProperties.Type.PROPERTY);
+    configuration = TEST_CONFIGS.get(testConfig);
+    if (configuration == null) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Unknown test configuration: [%s]. Known configs: %s",
+              testConfig, TEST_CONFIGS.keySet()));
+    }
+
+    // tempLocation needs to be set for bigtable IO writes
+    if (!Strings.isNullOrEmpty(tempBucketName)) {
+      String tempLocation = String.format("gs://%s/temp/", tempBucketName);
+      writePipeline.getOptions().as(TestPipelineOptions.class).setTempRoot(tempLocation);
+      writePipeline.getOptions().setTempLocation(tempLocation);
+      readPipeline.getOptions().as(TestPipelineOptions.class).setTempRoot(tempLocation);
+      readPipeline.getOptions().setTempLocation(tempLocation);
+    }
+  }
+
+  @AfterClass
+  public static void tearDownClass() {
+    ResourceManagerUtils.cleanResources(resourceManager);
+  }
+
+  private static final Map<String, Configuration> TEST_CONFIGS =
+      ImmutableMap.of(
+          "local", Configuration.of(20_000L, 5, "DirectRunner"), // 1MB
+          "small", Configuration.of(2_000_000L, 40, "DataflowRunner"), // 100 MB
+          "medium", Configuration.of(20_000_000L, 40, "DataflowRunner"), // 1 GB
+          "large", Configuration.of(200_000_000L, 100, "DataflowRunner") // 10 GB
+          );
+
+  /** Run integration test with configurations specified by TestProperties. */
+  @Test
+  public void testWriteAndRead() throws IOException {
+    tableId = generateTableId(testName);
+    resourceManager.createTable(tableId, ImmutableList.of(COLUMN_FAMILY_NAME));
```

Comment:
createTable has a default maxAge of 1h. This causes cells without setting timestamp (default to epoch) gets garbage collected. I think it should be never by default. What do you think @pranavbhandari24

Reply:
I've updated the [javadoc](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/759/files#diff-8b8008b8be3d1d84f327e4408a9d975f10c33988b276143b964595aa83a706ecR55) to make it clearer, please check

Reply:
That's interesting. Yeah, I'd agree that the data should never be garbage collected (if that is an option)
Will look into this.